violet-framework/src/main/java/com/horstmann/violet/workspace/editorpart/behavior/CutCopyPasteBehavior.java

```diff
@@ -195,16 +193,76 @@ public void paste()
 
             editorPart.getSwingComponent().invalidate();
             editorPart.getSwingComponent().repaint();
+
+            validatePaste(nodes, graph);
         }
         catch (IOException e)
         {
-            // Nothing to do
         }
     }
 
+    private void validatePaste(List<INode> nodes, IGraph graph){
+        if(checkIfNumberOfInitialNodesIsMoreThanOne(nodes, graph))
+        {
+            showWarningAboutNumberOfInitialNodes();
+        }
+    }
+    /**
+     * Is checking for each copied node if there is too many initialNodes on workspace
+     *
+     * @param nodes
+     * @param graph
+     * @return true if number of initialNodes is more than one
+     */
+    private boolean checkIfNumberOfInitialNodesIsMoreThanOne(List<INode> nodes, IGraph graph)
+    {
+        Collection<INode> nodeCollection = graph.getAllNodes();
+        for (INode newNode : nodes)
+        {
+            if (isNumberOfInitialNodesIsToHigh(newNode, nodeCollection))
+            {
+                return true;
+            }
+        }
+        return false;
+    }
+    /**
+     * Is checking if Number of InitialNodes is greater than one
+     *
+     * @param newNode
+     * @param nodeCollection
+     * @return true if number of initialNodes is more than one
+     */
+    private boolean isNumberOfInitialNodesIsToHigh(INode newNode, Collection<INode> nodeCollection)
+    {
+        if (newNode instanceof IInitialStateNode)
+        {
+            int numberOfInitialNodes = 0;
+
+            for (INode node : nodeCollection)
+            {
+                if (node instanceof IInitialStateNode)
+                {
+                    numberOfInitialNodes++;
+                }
+            }
+            if (numberOfInitialNodes > maxNumberOfInitialNodes)
+            {
+                return true;
```

Comment:
Its fine this way because for which reason shall this loop even procced if i already know that there is more than one initialState? When i got information that is already one in the workspace than i don't have really reason to check the rest of nodes. Please notice that there could be a lot of them in workspace(!)